cmd/deploy.go

```diff
@@ -291,3 +287,41 @@ func setFlags(conf *clab.Config) {
 		conf.Mgmt.IPv6Subnet = v6
 	}
 }
+
+// countWorkers calculates the number workers used for the creation of links and nodes.
+// If a user provided --max-workers value it is used for both links and nodes workers.
+// If maxWorkers is not set then the workers are limited by the number of available CPUs when
+// number of nodes/links exceeds the number of available CPUs.
+func countWorkers(nodeCount, linkCount, maxWorkers uint) (nodeWorkers, linkWorkers uint, err error) {
+	// init number of Workers to the number of links and nodes
+	nodeWorkers = nodeCount
+	linkWorkers = linkCount
+
+	switch {
+	// if maxWorkers is not set, limit workers number by number of available CPUs
+	case maxWorkers <= 0:
+		// retrieve vCPU count
+		vCpus, err := numcpus.GetOnline()
+		if err != nil {
+			return 0, 0, err
+		}
+
+		numCPUs := uint(vCpus)
+
+		// limit node/link workers only if there is more node/links thans CPU cores available
+		if nodeCount > numCPUs {
```

Comment:
Hi @steiler 
thanks for having a stab at it

What do you think about this tiny rework I did in 8b4a3bfa?

It might be subjective, but I think a `switch` makes things really easy to understand here, instead of the original maxThreadsCount var that served two purposes (both real cores count, as well as served as a default value when max workers is specified)

Reply:
lgtm